src/core/components/Button/index.tsx

```diff
@@ -0,0 +1,20 @@
+import { ButtonProps } from "@mui/material/Button";
+
+import * as S from "./styles";
+
+type Props = ButtonProps & {
+    title: string;
+    fullWidth?: boolean;
+
+    variant?: "contained" | "outlined" | "text";
+
+    onClick?: React.MouseEventHandler<HTMLButtonElement> | undefined;
```

Comment:
```suggestion
    onClick?: React.MouseEventHandler<HTMLButtonElement>;
```

Quando a gente usa optional type, não há a necessidade de colocar `| undefined`

Reply:
Removi o undefined